lib/AST/ASTContext.cpp

```diff
@@ -761,10 +761,15 @@ ASTContext::~ASTContext() {
 }
 
 void ASTContext::SetPreModuleImportCallback(
-    std::function<void(llvm::StringRef ModuleName, bool IsOverlay)> callback) {
+    PreModuleImportCallbackPtr callback) {
   PreModuleImportCallback = callback;
 }
 
+ASTContext::PreModuleImportCallbackPtr
+ASTContext::GetPreModuleImportCallback() const {
+  return PreModuleImportCallback;
+}
```

Comment:
What do you think of this? To move responsibility of the null check (of the callback) out of the caller's hands.

```suggestion
void ASTContext::PreModuleImport(llvm::StringRef ModuleName, ModuleImportKind kind) const {
  if (PreModuleImportCallback)
    PreModuleImportCallback(ModuleName, kind);
}
```